### PR TITLE
Remove CircleCI config

### LINKED ISSUE #1687: Migrate CI to GitHub Actions
Also parallelize more tasks when possible.

### PR DESCRIPTION
**Pull request by remoun**
We have stopped CircleCI from building this repo, and removed the required check status.

Closes #1687.

### CODE REVIEW

**Review by iamalwaysuncomfortable — COMMENTED** [review]
Are all of the github CI tasks currently working %100 as necessary?

Also, thanks for doing this! This is quite the effort.

**Comment by remoun** [comment]
> Are all of the github CI tasks currently working %100 as necessary?

Yes AFAIK. I'm going to give it another week or two before we stop using CircleCI.

> Also, thanks for doing this! This is quite the effort.

Cheers!

**Review by jgreat — APPROVED** [review]

**Review by eranrund — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,831 +0,0 @@
...
-          command: |
-            cargo build --frozen --target "$HOST_TARGET_TRIPLE" -p libmobilecoin
-      - check-dirty-git
-      - when:
-          condition: { equal: [ << pipeline.git.branch >>, master ] }
-          steps: [ save-sccache-cache, save-cargo-cache, save-homebrew-cache ]
-      - post-build
-
-  # Build and test go code
-  build-and-test-go:
```
Comment (garbageslam) [inline .circleci/config.yml:762]: I guess this is where we were building and testing the go code before the move to GitHub actions

Comment (remoun) [reply .circleci/config.yml:762]: Oh, I see I ported `lint-go` and missed this. Sorry about that!

**Pull request merged**